src/test_runner.rs

```diff
@@ -107,10 +113,20 @@ pub struct Config {
     /// The default is 1_000_000, which can be overridden by setting the
     /// `PROPTEST_MAX_FLAT_MAP_REGENS` environment variable.
     pub max_flat_map_regens: u32,
+    /// Indicates how to determine the file to use for persisting failed test
+    /// results.
+    ///
+    /// See the docs of [`FailurePersistence`](enum.FailurePersistence.html)
+    /// for more information.
+    ///
+    /// The default is `FailurePersistence::WithMain("proptest-failures.txt")`.
```

Comment:
Could each `#[test]` be given its own file by default? Perhaps not... It is very frustrating that we have `module_path!()` but not `function!()` to determine the name of the function... Can the `proptest!` macro perhaps use the `ident` for the test-function name and `stringify!(..)` it and then append that?

Reply:
The main part of the reasoning for having all the tests use one file is that I found that I often have things like

```rust
proptest! {
  fn test_a(x in foo_strategy()) { ... }
  fn test_b(x in foo_strategy()) { ... }
}
```

and if something breaks one test, there's a reasonable chance it could also break the other test.

Maybe a happy medium would be one failures file per source file? E.g., from `src/foo/bar.rs` you'd get something like `src/foo/bar-proptest-failures.txt`. That would also solve the question of "where to put this one global file by default?".

Reply:
I think that's a very reasonable compromise and you can use `module_path!()` directly then =)

Tho I think it might be good not to put the file directly in the same folder as the module and instead have a special fixture folder, like so:

```
src/foo/bar.rs => src/regressions/foo/bar-proptest-failures.txt
```
or:
```
src/foo/bar.rs => regressions/foo/bar-proptest-failures.txt
```